scipy/stats/_continuous_distns.py

```diff
@@ -6942,6 +6942,161 @@ def _stats(self, a):
+        that the shape is equal to or greater than one. If either location or
+        scale is fixed, then the other can be determined analytically. If only
+        the shape is fixed, or no parameters are fixed, then use `root_scalar`
+        to solve a system of equations formed by setting the partial
+        derivatives of the log-likelihood function with respect to the location
+        the the scale equal to each other. Location is used as the dependent
+        variable and then the shape and scale are analytically determined.
+
+        In many cases, the use of `np.nextafter` is utilized to prevent
+        roundoff error that causes the log likelihood equation to be very
+        large.
+        '''
+        data, fshape, floc, fscale = _check_fit_input_parameters(self, data,
+                                                                 args, kwds)
+        # ensure that any fixed parameters don't violate constraints of the
+        # distribution before continuing. The support of the distribution
+        # is `0 < (x - loc)/scale < 1`.
+        if floc is not None:
+            if not data.min() > floc:
+                raise FitDataError('powerlaw', 0, 1)
+            if fscale is not None and not data.max() <= floc + fscale:
+                raise FitDataError('powerlaw', 0, 1)
+
+        if fscale is not None:
+            if fscale <= 0:
+                raise ValueError("Negative or zero `fscale` is outside the "
+                                 "range allowed by the distribution.")
+            if fscale <= data.ptp():
+                msg = "`fscale` must be greater than the range of data."
+                raise ValueError(msg)
+
+        def get_shape(data, loc, scale):
+            # The first-order necessary condition on `shape` can be solved in
+            # closed form. It can be used whether no matter the assumption
+            # of the value of the shape.
+            return -len(data) / np.sum(np.log((data - loc)/scale))
+
+        def get_scale(data, loc):
+            # analytical solution for `scale` based on the location.
+            # It can be used whether no matter the assumption of the value of
+            # the shape.
+            return data.max() - loc
+
+        if fscale is not None:
+            # in the case where the scale is set to a non-optimal value, there
+            # may be two possible analytical solutions, but one gives a better
+            # log-likelihood result.
+            args = [data, (self._fitstart(data),)]
+            func = self._reduce_func(args, {})[1]
+
+            # using `data.min()` as the optimal location
+            loc_lt1 = np.nextafter(data.min(), -np.inf)
+            shape_lt1 = fshape or get_shape(data, loc_lt1, fscale)
+            ll_lt1 = func((shape_lt1, loc_lt1, fscale), data)
+
+            # using `data.max() - scale` as the optimal location
+            loc_gt1 = np.nextafter(data.max() - fscale, np.inf)
```

Comment:
I think this logic is broken, since it doesn't even check to see if `floc` is set, and then returns _always_ when `fscale` is set.